Framework/PythonInterface/mantid/api/_workspaceops.py

```diff
@@ -245,18 +246,9 @@ def _method_impl(self, *args, **kwargs):
         return func_obj(*args, **kwargs)
 
     # ------------------------------------------------------------------
-    # Add correct meta-properties for the method
-    if hasattr(func_obj, '__signature__'):
-        from inspect import Parameter
-        func_parameters = list(func_obj.__signature__.parameters.values())
-        func_parameters.insert(0, Parameter("self", Parameter.POSITIONAL_ONLY))
-        signature = func_obj.__signature__.replace(parameters=func_parameters)
-    else:
-        signature = ['self']
-        signature.extend(getsource(func_obj).co_varnames)
-        signature = tuple(signature)
+    from mantid.simpleapi import LazyMethodSignature  # Import here to avoid circular depandancy.
```

Comment:
Can we move the `LazyMethodSignature` & `LazyFunctionSignature` classes to `mantid.kernel.funcinspect` rather than having everything in `simpleapi`?